### DISCUSSION TITLE
Relation between Content, Assessment and Questions/QuestionSet

### DISCUSSION THREAD

**Discussion by Debayanpaul1593** [Sunbird-Knowlg/Community#120] [General Discussions]
I am working on a new requirement to measure Content progress based on Assessments completed:

1. Firstly if someone could clarify if an assessment is a question/questionSet itself or a question/questionSet linked to a Content?
2. When we create a QuestionSet hierarchy and want it to be part of a Course Collection(CC) hierarchy do we add the QuestionSet directly to the CC hierarchy or do we first attach it to a content of primaryCategory "Course Assessment"  and then add it to CC hierarchy? If the latter is possible then how do we make the connection between Content and QuestionSet?
3. In sunbird-course-service on submitting a course assessment for updateCourseConsumption `/v1/content/state/update`, the assessment is hardcoded a status value of 2 which gets translated to a progress of 100 at the time of cassandra db write(keyspace: **sunbird_courses**, table: **user_content_consumption** ). Is this something the knowlg team can help with or do I have to raise with Lern Team?

**Comment by Debayanpaul1593** [comment]
Follow up on the discussion:
In content/1/0/config.json one of the relations is defined as 
`"relations": {
        "questions": {
            "type": "associatedTo",
            "direction": "out",
            "objects": ["AssessmentItem"]
        },
}`
What is 'AssessmentItem' here? There is no controller for one so how to create it?
If I create a content with 'questions' relation and pass an actual question or questionSet identifier then I get the following error:
`"errmsg": "Error while validating relations :: List(associatedTo is not allowed between Content and Question)"`

**Comment by pallakartheekreddy** [comment]
@Debayanpaul1593 
1. Firstly if someone could clarify if an assessment is a question/questionSet itself or a question/questionSet linked to a Content?
>  Currently, in sunbird we can create two types of assessments one is create questions using content editor and select multiple questions and create an assessment content(ECML content) and second is using QuML editor you can create questionSet hierarchy and create QuML questions under it. 

2. When we create a QuestionSet hierarchy and want it to be part of a Course Collection(CC) hierarchy do we add the QuestionSet directly to the CC hierarchy or do we first attach it to a content of primaryCategory "Course Assessment" and then add it to CC hierarchy? If the latter is possible then how do we make the connection between Content and QuestionSet?
> Yes, You can create a QuestionSet hierarchy and add that directly in course there is no restriction collection hierarchy API. But `assessment-aggregator` job which will calculate progress for assements based on primaryCategory i think curretly is is set as `Course Assessment` please update that config with QuestionSet primaryCategory as well. So progress will calculate seems less
@AmiableAnil @Jayaprakash8887 please add more details.

3. In sunbird-course-service on submitting a course assessment for updateCourseConsumption `/v1/content/state/update`, the assessment is hardcoded a status value of 2 which gets translated to a progress of 100 at the time of cassandra db write(keyspace: sunbird_courses, table: user_content_consumption ). Is this something the knowlg team can help with or do I have to raise with Lern Team?
>  @Debayanpaul1593 Please raise this query in lern discussion forum. https://github.com/orgs/Sunbird-Lern/discussions

**Reply by pallakartheekreddy** [reply]
@Debayanpaul1593 Pls let us know if your query is resolved

**Reply by Debayanpaul1593** [reply]
Yes @pallakartheekreddy the query is resolved.
